src/resmom_win/start_exec.c

```diff
@@ -687,23 +732,25 @@ bld_wenv_variables(char *name, char *value)
 
 	if (env_array == NULL) {
 		numenv = 32;
-		env_array = (char **)malloc(sizeof(char *) * numenv);
+		env_array = (char **)malloc(sizeof(char *) * (numenv+1));
+					 /* +1 for the sentinel (end) entry */
 		if (env_array == NULL) {
 			log_err(errno, "bld_wenv_variables", "failed to malloc");
 			return;
 		}
 		/* initialize the unused entries to NULL */
 		for (i=0; i < numenv; i++)
 			env_array[i] = NULL;
+		env_array[i] = NULL; /* sentinel (end) entry */
 	}
 	/* curenv is the index to th next slot that can be used in env_array. */
 	/* numenv is the number of elements currently allocated in env_array. */
-	if (curenv == numenv) {		/* none left */
+	if (curenv >= numenv) {		/* none left */
 		int	numenv_tmp = 0;
 		char	**env_array_tmp = NULL;
 
 		numenv_tmp = numenv * 2;
-		env_array_tmp = (char **)realloc(env_array, sizeof(char *)*numenv_tmp);
+		env_array_tmp = (char **)realloc(env_array, sizeof(char *)*(numenv_tmp+1));	/* +1 for sentinel (end) entry */
```

Comment:
Please add space around the `*` in `sizeof(char *)*(numenv_tmp+1)`

Reply:
Done.